boot/nuttx/src/flash_map_backend/flash_map_backend.c

```diff
@@ -65,7 +65,7 @@ struct flash_device_s
   struct partition_info_s partinfo;
 
   int     fd;          /* File descriptor for an open flash area */
-  int32_t refs;        /* Reference counter */
+  int     refs;        /* Reference counter */
```

Comment:
I believe the proper fix lies on the log format rather than changing the data type here.
Please, consider maintaining the fixed-width type and fixing the formats where applicable.

Reply:
I really do not think that it matters. The MCUboot is "Secure boot for 32-bit Microcontrollers", so `int` should be capable of capable of carrying same values as `int32_t`. I chose a single change head instead of dozen of changes in log calls because fixes seems equivalent to me.

Reply:
> I really do not think that it matters. The MCUboot is "Secure boot for 32-bit Microcontrollers", so int should be capable of capable of carrying same values as int32_t.

This is a matter of consistency. In general it is more recommended to rely on fixed-width data types, which makes the overall code clearer and less prone to portability issues.

> I chose a single change head instead of dozen of changes in log calls because fixes seems equivalent to me.

I didn't find those dozen of log calls, there only 3 references that you'd need to change:
https://github.com/mcu-tools/mcuboot/blob/main/boot/nuttx/src/flash_map_backend/flash_map_backend.c#L266-L267
https://github.com/mcu-tools/mcuboot/blob/main/boot/nuttx/src/flash_map_backend/flash_map_backend.c#L322
https://github.com/mcu-tools/mcuboot/blob/main/boot/nuttx/src/flash_map_backend/flash_map_backend.c#L366-L367

Reply:
> This is a matter of consistency. In general it is more recommended to rely on fixed-width data types, which makes the overall code clearer and less prone to portability issues.
I understand your point, but taking into account the definition of MCUboot itself as a "32-bit Microcontrollers" variant I do not see portability issue as a strong argument.
> I didn't find those dozen of log calls, there only 3 references that you'd need to change:
Yeah, probably the warning log was too long so I didn't count all places, but rather saw that a single line change will fix all.

Anyway I will proceed and apply 3 changes instead of one.

Reply:
> I understand your point, but taking into account the definition of MCUboot itself as a "32-bit Microcontrollers" variant I do not see portability issue as a strong argument.

Just because MCUboot only supports 32-bit microcontrollers today does not mean it will always remain like that. That's why I said "**less prone to** portability issues".

> Anyway I will proceed and apply 3 changes instead of one.

Amount of changed lines should not matter as long as the solution is the most appropriate.

Reply:
Done. Please check code consistency improvements